xfer/model_handler/model_handler.py

```diff
@@ -43,103 +44,498 @@ def __init__(self, module, context_function=mx.context.cpu, num_devices=1, data_
+            # If case is not ambiguous then the dropped layer is simply the first layer
+            if drop_layer_name is None:
+                drop_layer_name = self._get_name_of_first_operation(symbol_dict[consts.NODES])
+
+            logging.info('Dropping {}'.format(drop_layer_name))
+            layers_dropped.append(drop_layer_name)
+
+            # Find node index of operator being deleted
+            del_node_op_idx = self._get_layer_node_idx(symbol_dict[consts.NODES], drop_layer_name)
+            symbol_dict = self._delete_layer_nodes_given_operator_node(symbol_dict, del_node_op_idx)
+
+            symbol_dict, join_idx, join_deleted, join_layer_name = self._remove_redundant_join_layer(
+                                                                    symbol_dict=symbol_dict,
+                                                                    drop_layer_name=drop_layer_name,
+                                                                    nodes_before=temp_symbol_dict[consts.NODES],
+                                                                    deleted_node_operator_idx=del_node_op_idx)
+            if join_deleted:
+                layers_dropped.append(join_layer_name)
+
+            # Update symbol dictionary attributes
+            symbol_dict[consts.ARG_NODES] = self._get_arg_nodes(symbol_dict[consts.NODES])
+            symbol_dict[consts.HEADS] = self._get_heads(nodes=symbol_dict[consts.NODES],
+                                                        output_layer_names=self._get_output_layer_names(
+                                                            nodes=temp_symbol_dict[consts.NODES],
+                                                            heads=temp_symbol_dict[consts.HEADS]))
+            symbol_dict[consts.NODES] = self._update_inputs(nodes=symbol_dict[consts.NODES],
+                                                            original_nodes=temp_symbol_dict[consts.NODES],
+                                                            drop_layer_name=drop_layer_name,
+                                                            join_deleted=join_deleted,
+                                                            join_idx=join_idx)
+
+        sym = mx.sym.load_json(json.dumps(symbol_dict))
+
+        logging.info('{} deleted from model bottom'.format(', '.join(layers_dropped)))
+        if drop_layer_names is not None:
+            if len(drop_layer_names) > 0:
+                logging.warning('Did not use all of drop_layer_names: {}'.format(', '.join(drop_layer_names)))
         self.update_sym(sym)
 
-    def add_layer_top(self, layer_factory_list):
+    def add_layer_top(self, layer_list):
         """
         Add layer to output of model.
 
-        :param layer_factory: List of LayerFactory objects to be added to model output.
-        :type layer_factory: list(:class:`LayerFactory`)
+        :param layer_list: List of MxNet symbol layers to be added to model output.
```

Comment:
Document the order in which layers are added to the top

Reply:
Added a line in the method docsting explaining the ordering. 163-164